### ISSUE TITLE
CostMap should contain all map information

### ISSUE THREAD

**Issue by yoos** [osurdml/issues#17]
CostMap currently stores just cost values in a matrix. Along with the current state, it is used as an input to the exploration algorithm to generate a set of possible future states (i.e., possible actions to take).

The algorithm will need other information such as depth at every map location in addition to the cost. I propose that CostMap be renamed and modified to contain this information.

**Comment by yoos** [comment]
I agree.

**Comment by drews256** [comment]
I'm assuming this is the matlab code? Please feel free to manipulate the variable names. Everything was a work in progress when I wrote it. But i was hoping that cost would just be a matrix of the cost of being at that point. Without the distance or anything else added in. It could be renamed as the specific cost...or something more interesting.

**Comment by yoos** [comment]
No, CostMap is a C++ class I've put together for use in treemower. Currently, it is indeed just a matrix of the cost at every geographical point. I thought since it used to be passed to the exploration algorithm as an input, I might as well include information about other features of the map.

That said, I've since done some refactoring that makes it easy to simply create another class for the other information, especially as they'll likely be used differently.

**Comment by drews256** [comment]
Ahh. Ok. So without saying it. You basically did what I was thinking. Good work. Imma probably being working on this stuff next week. But it'll be next week for sure.

**Comment by yoos** [comment]
Frabjous!

**Comment by yoos** [comment]
After some discussion with Yawei, I've realized the expected use of what we're calling a "cost" map is not what I'd thought it was: the cost of traversing through a location on the map.

The map Yawei (presumably Drew, as well) has been using is a map of _scores_ representing the utility of collecting data at a map location, i.e., a scoremap. Since we are working with a surface water vehicle, the cost of transport per given distance is assumed to be unrelated to the actual location.

This doesn't change the fact that we need another container (ScoreMap?) to store sensory data and other information used in making navigation decisions, but it's important to note that CostMap in the way I had imagined it currently has no planned use.
